research/research.py

```diff
@@ -179,22 +182,17 @@ def upload_post() -> Response:
     chunk_data = request.files['file']
     encode_unicode_content = iRODSMessage.encode_unicode(chunk_data.stream.read())
 
+    # Write chunk data.
     try:
-        if flow_chunk_number == 1:
-            with session.data_objects.open(object_path, 'w') as obj_desc:
-                obj_desc.write(encode_unicode_content)
-
-            obj_desc.close()
-        else:
-            with session.data_objects.open(object_path, 'a') as obj_desc:
-                obj_desc.seek(int(flow_chunk_size * (flow_chunk_number - 1)))
-                obj_desc.write(encode_unicode_content)
-
-            obj_desc.close()
+        with session.data_objects.open(object_path, 'a', rescName=app.config.get('IRODS_DEFAULT_RESC')) as obj_desc:
+            obj_desc.seek(int(flow_chunk_size * (flow_chunk_number - 1)))
+            obj_desc.write(encode_unicode_content)
     except Exception:
         response = make_response(jsonify({"message": "Chunk upload failed"}), 500)
         response.headers["Content-Type"] = "application/json"
         return response
+    finally:
+        obj_desc.close()
```

Comment:
This line results in traces in the error log when obj_desc is referenced before assignment. This could be avoided by either testing if obj_desc has been assigned, or by catching and ignoring any exceptions as a result of this line.